src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -741,9 +796,22 @@ public unsafe virtual int IndexOf(String source, String value, int startIndex, i
             if ((options & ValidIndexMaskOffFlags) != 0 && (options != CompareOptions.Ordinal))
                 throw new ArgumentException(SR.Argument_InvalidFlag, nameof(options));
 
+            if (_invariantMode)
+                return IndexOfOrdinal(source, value, startIndex, count, ignoreCase: (options & (CompareOptions.IgnoreCase | CompareOptions.OrdinalIgnoreCase)) != 0);
+
             return IndexOfCore(source, value, startIndex, count, options, null);
         }
 
+        internal static unsafe int IndexOfOrdinal(string source, string value, int startIndex, int count, bool ignoreCase)
+        {
+            if (GlobalizationMode.Invariant)
```

Comment:
Move this to the caller in this file, so that it can use the faster instance check?